Source/Scene/GroundPrimitive.js

```diff
@@ -0,0 +1,762 @@
+            backOperation : {
+                fail : StencilOperation.KEEP,
+                zFail : StencilOperation.KEEP,
+                zPass : StencilOperation.DECREMENT_WRAP
+            },
+            reference : 0,
+            mask : ~0
+        },
+        depthTest : {
+            enabled : false
+        },
+        depthMask : false,
+        blending : BlendingState.ALPHA_BLEND
+    };
+
+    var pickRenderState = {
+        stencilTest : {
+            enabled : true,
+            frontFunction : StencilFunction.NOT_EQUAL,
+            frontOperation : {
+                fail : StencilOperation.KEEP,
+                zFail : StencilOperation.KEEP,
+                zPass : StencilOperation.DECREMENT_WRAP
+            },
+            backFunction : StencilFunction.NOT_EQUAL,
+            backOperation : {
+                fail : StencilOperation.KEEP,
+                zFail : StencilOperation.KEEP,
+                zPass : StencilOperation.DECREMENT_WRAP
+            },
+            reference : 0,
+            mask : ~0
+        },
+        depthTest : {
+            enabled : false
+        },
+        depthMask : false
+    };
+
+    var scratchBVCartesianHigh = new Cartesian3();
+    var scratchBVCartesianLow = new Cartesian3();
+    var scratchBVCartesian = new Cartesian3();
+    var scratchBVCartographic = new Cartographic();
+    var scratchBVRectangle = new Rectangle();
+
+    function createBoundingVolume(primitive, frameState, geometry) {
+        var highPositions = geometry.attributes.position3DHigh.values;
+        var lowPositions = geometry.attributes.position3DLow.values;
+        var length = highPositions.length;
+
+        var ellipsoid = frameState.mapProjection.ellipsoid;
+
+        var minLat = Number.POSITIVE_INFINITY;
+        var minLon = Number.POSITIVE_INFINITY;
+        var maxLat = Number.NEGATIVE_INFINITY;
+        var maxLon = Number.NEGATIVE_INFINITY;
+
+        for (var i = 0; i < length; i +=3) {
+            var highPosition = Cartesian3.unpack(highPositions, i, scratchBVCartesianHigh);
+            var lowPosition = Cartesian3.unpack(lowPositions, i, scratchBVCartesianLow);
+
+            var position = Cartesian3.add(highPosition, lowPosition, scratchBVCartesian);
+            var cartographic = ellipsoid.cartesianToCartographic(position, scratchBVCartographic);
+
+            var latitude = cartographic.latitude;
+            var longitude = cartographic.longitude;
+
+            minLat = Math.min(minLat, latitude);
+            minLon = Math.min(minLon, longitude);
+            maxLat = Math.max(maxLat, latitude);
+            maxLon = Math.max(maxLon, longitude);
+        }
+
+        var rectangle = scratchBVRectangle;
+        rectangle.north = maxLat;
+        rectangle.south = minLat;
+        rectangle.east = maxLon;
+        rectangle.west = minLon;
+
+        var obb = OrientedBoundingBox.fromRectangle(rectangle, GroundPrimitive._maxHeight, GroundPrimitive._minOBBHeight, ellipsoid);
```

Comment:
Also add a TODOs for
* Dynamic min height in the vertex shader
* A tighter max height
* _Maybe_ batching by color